example/locked_cout.hpp

```diff
@@ -0,0 +1,46 @@
+// Copyright Takatoshi Kondo 2019
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(MQTT_LOCKED_COUT_HPP)
+#define MQTT_LOCKED_COUT_HPP
+
+#include <mutex>
+#include <iostream>
+
+class locked_stream {
+public:
+    locked_stream(std::ostream& stream)
+        :lock_(mtx_),
+         stream_(stream) {}
+
+    friend
+    locked_stream&& operator<<(locked_stream&& s, std::ostream& (*arg)(std::ostream&)) {
```

Comment:
why the use of rvalue references here?

Reply:
This is to write as follows:

```cpp
locked_cout() << data1 << data2;
```

locked_cout() is rvalue.